packages/sources/coinmetrics-test/src/transports/totalBurnedTransport.ts

```diff
@@ -0,0 +1,208 @@
+import { AdapterConfig } from '@chainlink/external-adapter-framework/config'
+import { RestTransport } from '@chainlink/external-adapter-framework/transports'
+import {
+  AdapterRequest,
+  AdapterResponse,
+  makeLogger,
+  sleep,
+} from '@chainlink/external-adapter-framework/util'
+import { axiosRequest } from '@chainlink/external-adapter-framework/transports/util'
+import * as rateLimitMetrics from '@chainlink/external-adapter-framework/rate-limiting/metrics'
+import { ethers, BigNumber } from 'ethers'
+import { AdapterDataProviderError } from '@chainlink/external-adapter-framework/validation/error'
+import { customSettings } from '../config'
+
+const logger = makeLogger('Coinmetrics TotalBurnedTransport')
+
+export interface TotalBurnedInputParams {
+  asset: string
+  frequency: string
+  pageSize: number
+  startTime: string
+  endTime: string
+}
+
+export interface AssetMetrics {
+  asset: string
+  time: string
+  FeeTotNtv: string
+  IssTotNtv: string
+  RevNtv: string
+}
+
+export interface ResponseSchema {
+  data: AssetMetrics[]
+  next_page_token?: string
+  next_page_url?: string
+  totalBurnedTKN: BigNumber
+}
+
+export interface ProviderRequestBody {
+  asset: string
+  frequency: string
+  pageSize: number
+  startTime: string
+  endTime: string
+  isBurnedEndpointMode: false
+}
+
+// Common frequencies for FeeTotNtv, RevNtv and IssTotNtv metrics
+export enum Frequency {
+  ONE_DAY = '1d',
+  ONE_BLOCK = '1b',
+}
+
+export const METRICS = 'FeeTotNtv,RevNtv,IssTotNtv'
+export const DEFAULT_PAGE_SIZE = 10_000
+export const URL = 'timeseries/asset-metrics'
+
+export type TotalBurnedEndpointTypes = {
+  Request: {
+    Params: TotalBurnedInputParams
+  }
+  Response: {
+    Data: ResponseSchema | undefined
+    Result: string
+  }
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: ProviderRequestBody
+    ResponseBody: ResponseSchema
+  }
+}
+
+export const calculateBurnedTKN = (assetMetricsList: AssetMetrics[]): BigNumber => {
+  let burnedTKN = BigNumber.from('0')
+  assetMetricsList.forEach((assetMetrics: AssetMetrics) => {
+    let feeTotNTV
+    let revNtv
+    let issTotNtv
+    try {
+      feeTotNTV = ethers.utils.parseEther(assetMetrics.FeeTotNtv)
+      revNtv = ethers.utils.parseEther(assetMetrics.RevNtv)
+      issTotNtv = ethers.utils.parseEther(assetMetrics.IssTotNtv)
+    } catch (error) {
+      throw new Error(
+        `Unprocessable asset metrics: ${JSON.stringify(assetMetrics)}, due to: ${error}.`,
+      )
+    }
+    burnedTKN = burnedTKN.add(feeTotNTV.sub(revNtv.sub(issTotNtv)))
+  })
+  return burnedTKN
+}
+
+export class TotalBurnedTransport extends RestTransport<TotalBurnedEndpointTypes> {
+  async foregroundExecute(
+    req: AdapterRequest<TotalBurnedEndpointTypes['Request']>,
+    config: AdapterConfig<TotalBurnedEndpointTypes['CustomSettings']>,
```

Comment:
Are the explicit types necessary?

Reply:
Due to the custom request loop in this transport, we need to pull a few values from the DP response which are specific to `TotalBurnedTransport`, so if we don't make the types explicit at the top, then we have to cast the DP response to an explicit value below. Do you think it's better to do the casting inside the `foregroundExecute` method?